lib/rules-generator.js

```diff
@@ -218,25 +251,34 @@ util.methods(Generator, {
     var i;
     var pathMethods = ['read', 'write'];
     var location = this.ensurePath(path.parts);
+    var exp;
 
-    // Write validation (if non-trivial)
-    location['.validate'] = this.getExpressionText(
-      this.partialEval(this.symbols.functions['@validator@' + path.isType].body));
-    if (location['.validate'] == 'true') {
-      delete location['.validate'];
+    // Path validation function && Type Validation
+    var terms = [];
+    if (path.methods.validate) {
+      terms.push(path.methods.validate.body);
+    }
+    terms.push(this.symbols.functions['@validator@' + path.isType].body);
+
+    exp = this.partialEval(ast.andArray(terms));
+    if (!(exp.type == 'Boolean' && exp.value == true)) {
+      location['.validate'] = this.getExpressionText(exp);
     }
 
     // Write read and write methods
     for (i = 0; i < pathMethods.length; i++) {
```

Comment:
This is a bit convoluted for what its doing. Too much hunting around source to realise you method ends up as either "read" or "write". What about for ["read", "write"]).map {function(method) {}}